src/datarecord.cpp

```diff
@@ -588,6 +588,24 @@ void datarecord::schedule(reclist& thisi, double maxtime,
   }
 }  
 
+/*  Insert observation records triggered from evtools
+ * 
+ */
+
+void insert_observations(reclist& thisi, mrgsolve::evdata& ev, const size_t start, 
+                         const bool put_ev_first) {
+  const int total = ev.addl + 1;
```

Comment:
Notice that we're incrementing `addl` here, to insert the equivalent number of records as we would for a dose.